openpype/hosts/traypublisher/plugins/publish/collect_simple_instances.py

```diff
@@ -65,6 +68,12 @@ def process(self, instance):
             representation_files_mapping
         )
 
+        # Hack to set env vars required to run in the farm
+        os.environ["AVALON_ASSET"] = instance.data["asset"]
+        os.environ["AVALON_TASK"] = instance.data.get("task")
+        legacy_io.Session["AVALON_ASSET"] = instance.data["asset"]
+        legacy_io.Session["AVALON_TASK"] = instance.data.get("task")
+
```

Comment:
Why?

Reply:
it's required so it gets written in the json file that's used in the headless publish so the task can be collected in the farm. The purpose of this PR being a draft though is to start discussion to figure out if there's a better way of doing this